common/src/main/java/com/revenuecat/purchases/common/networking/MapConverter.kt

```diff
@@ -0,0 +1,58 @@
+package com.revenuecat.purchases.common.networking
+
+import org.json.JSONArray
+import org.json.JSONObject
+
+/**
+ * A class to convert a Map<String, Any?> into a JSONObject.
+ * This was created to workaround a bug in Android 4 , where a List<String> would be incorrectly converted into
+ * a single string instead of a JSONArray of strings. (i.e.: "[\"value1\", \"value2\"]" instead of "[value1, value2]")
+ * This class handles nested maps, lists, and other JSON-compatible types.
+ */
+class MapConverter {
+
+    /**
+     * Converts the given [inputMap] into a JSONObject.
+     *
+     * @param inputMap The input map to convert.
+     * @return A JSONObject representing the input map.
+     */
+    internal fun convertToJSON(inputMap: Map<String, Any?>): JSONObject {
+        val mapWithoutInnerMaps = inputMap.mapValues { (_, value) ->
+            value.tryCast<Map<String, Any?>>(ifSuccess = { convertToJSON(this) })
```

Comment:
Unless I'm reading this wrong, this line doesn't do anything. It will convert to JSON but we are ignoring the return value I think? We moved this line to the `else` inside the `when`, so I think this line can be removed if I'm not wrong.

Reply:
yes, I was thinking the same

Reply:
oops, leftover from refactor, thanks